features/extra/helpers.js

```diff
@@ -40,16 +40,15 @@ module.exports = {
     if (!options.backoff) options.backoff = 500;
     if (!options.maxTime) options.maxTime = 5;
 
-    const delay = options.delay;
```

Comment:
This function has likely been broken for a very long time. `delay += options.backoff;` could never have worked with `const delay`. It's likely the `AWS.util.date.getDate()` call did not return a reasonable value and the retry did not happen, meaning previously passing e2e tests always succeeded on the first try of the `next.condition`.

Reply:
The `AWS.util.date.getDate()` is a remnant from AWS SDK for JavaScript (v2), which wouldn't have worked in v3.
Thanks for catching it!